src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -70,23 +72,50 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
         {
             appModel.Resources.Add(additionalModule);
         }
+
+        static void SetKnownParameterValue(AzureBicepResource r, string key, Func<AzureBicepResource, object> factory)
+        {
+            if (r.Parameters.TryGetValue(key, out var existingValue) && existingValue is null)
+            {
+                var value = factory(r);
+
+                if (value is BicepOutputReference b && b.Resource == r)
+                {
+                    return;
+                }
```

Comment:
What is this check for? Can we add a comment here?

It would be odd that a parameter is pointing to an output of the same resource, isn't it?

Reply:
I'll add a comment. It really is an odd case (it was because we're using principalId)

Reply:
Removed!